Source/Core/Common/HttpRequest.h

```diff
@@ -11,6 +11,8 @@
 #include <string>
 #include <vector>
 
+#include <curl/curl.h>
```

Comment:
Please revert this and the callback change. This causes curl to leak through and defeats the entire purpose of having an abstraction.

Reply:
Probably instead of using `curl_off_t`, `ProgressCallback` could be changed to use `s64` (keeping the change in HttpRequest.cpp).

Reply:
I hope the new solution I came up with is satisfactory.